cht/adapter.py

```diff
@@ -2,307 +2,296 @@
-                self.writeHCoeffDataID = self.precice.getDataID(writeDataName, self.preciceNodesMeshID)
-            elif writeDataName.find("Sink-Temperature-") >= 0:
-                self.writeTempDataID = self.precice.getDataID(writeDataName, self.preciceNodesMeshID)
-        for readDataName in names["read-data"]:
-            if readDataName.find("Heat-Transfer-Coefficient-") >= 0:
-                self.readHCoeffDataID = self.precice.getDataID(readDataName, self.preciceFaceCentersMeshID)
-            elif readDataName.find("Sink-Temperature-") >= 0:
-                self.readTempDataID = self.precice.getDataID(readDataName, self.preciceFaceCentersMeshID)
-
-    def getPreciceNodeIndices(self):
-        return self.preciceNodeIndices
-    
-    def getPreciceFaceCenterIndices(self):
-        return self.preciceFaceCenterIndices
-    
-    def getPreciceNodesMeshID(self):
-        return self.preciceNodesMeshID
-    
-    def getPreciceFaceCentersMeshID(self):
-        return self.preciceFaceCentersMeshID
-    
-    def getNodeCoordinates(self):
-        return self.nodeCoordinates
-    
-    def getFaceCenterCoordinates(self):
-        return self.faceCenterCoordinates
-        
-    def getNormals(self):
-        return self.normals
-    
-    def createBCs(self):
-        """
+	def __init__(self, precice, participantName, names, MESH, SHMESH, MODEL, MAT, isNonLinear = False):
+
+		self.precice = precice
+		self.participantName = participantName
+
+		self.groupName = ""
+		self.facesMeshName = ""
+		self.nodesMeshName = ""
+
+		self.nodes = []
+		self.faces = []
+		self.connectivity = []
+		self.nodeCoordinates = []
+		self.faceCenterCoordinates = []
+		self.normals = None
+
+		self.isNonLinear = isNonLinear
+		self.conductivity = None
+		self.isConductivityInitialized = False
+		self.delta = 1e-5
+
+		self.preciceNodeIndices = []
+		self.preciceFaceCenterIndices = []
+
+		self.preciceFaceCentersMeshID = 0
+		self.preciceNodesMeshID = 0
+
+		self.readHCoeffDataID = 0
+		self.readTempDataID = 0
+		self.writeHCoeffDataID = 0
+		self.writeTempDataID = 0
+
+		self.readTemp = []
+		self.readHCoeff = []
+		self.writeTemp = []
+		self.writeHCoeff = []
+
+		self.readDataSize = 0
+		self.writeDataSize = 0
+
+		self.MESH = MESH
+		# Shifted mesh (contains only the interface, and is shifted by delta in the direction opposite to the normal)
+		self.SHMESH = SHMESH
+		self.MODEL = MODEL
+		self.MAT = MAT
+		self.LOAD = None
+		self.LOADS = []
+		self.mesh = partition.MAIL_PY()
+		self.mesh.FromAster(MESH)
+
+		self.configure(names)
+
+	def configure(self, names):
+
+		self.groupName = names["patch"]
+
+		# In Code_Aster, write-data is located at the nodes
+		self.nodesMeshName = names["write-mesh"]
+		# and read-data is located at the face centers
+		self.faceCentersMeshName = names["read-mesh"]
+
+		self.computeNormals()
+
+		self.nodeCoordinates = np.array([p for p in self.SHMESH.sdj.COORDO.VALE.get()])
+		# Casted len(...)/3 to int
```

Comment:
You can remove this comment now.

Reply:
After discussing, I see the problem with rounding. Please fix this before merging.

Reply:
In the original code (python 2.x) the division is done with an "integer division", meaning that the result of x/3 will be floored. In python3 the division of two integers is a true division, returning a double. To get the desired outcome, an additional casting to `int` is necessary after the division operation.
The comments will be removed.